src/extension.c

```diff
@@ -281,6 +283,20 @@ ts_extension_is_loaded(void)
 			 * that, for example, the catalog does not go looking for things
 			 * that aren't yet there.
 			 */
+			if (extstate == EXTENSION_STATE_TRANSITIONING)
+			{
+				/* when we are updating the extension, we execute
+				 * scripts in post_update.sql after setting up the
+				 * the dependencies. At this stage, TS
+				 * specific functionality is permitted as we now have
+				 * all catalogs and functions in place
+				 */
+				const char *update_script_stage =
+					GetConfigOption(TS_UPDATE_SCRIPT_CONFIG_VAR, true, false);
+				if (update_script_stage &&
+					(strncmp(update_script_stage, POST_UPDATE, strlen(POST_UPDATE)) == 0))
```

Comment:
this code checks that update_script_stage starts with the same chars as POST_UPDATE but is not required to be equal, is this intended?

Reply:
added equality test